core-metrics/src/main/java/org/apache/kylin/metrics/QuerySparkMetrics.java

```diff
@@ -0,0 +1,331 @@
+package org.apache.kylin.metrics;
+
+import org.apache.kylin.shaded.com.google.common.collect.Maps;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentMap;
+import java.util.stream.Collectors;
+
+public class QuerySparkMetrics {
+    private static Map<String, List<SparkJobMetrics>> sparkJobMetricsMap = Maps.newConcurrentMap();
+    private static List<SparkStageMetrics> sparkStageMetricsList = new ArrayList<>();
```

Comment:
Use a static collection is not a good practice . Maybe using a `BlockingQueue` is better, like [Producer-Consumer pattern](https://blog.csdn.net/qq_35580883/article/details/78650677) , or send directly to `AbstractActiveReservoir` is also good to me.